src/local_scheduler/local_scheduler_algorithm.cc

```diff
@@ -42,6 +42,10 @@ struct ObjectEntry {
    *  to the corresponding task's queue entry in waiting queue, for fast
    *  deletion when all of the task's dependencies become available. */
   std::vector<std::list<TaskQueueEntry>::iterator> dependent_tasks;
+  /** Whether or not to request a transfer of this object. This should be set
+   *  to true for all objects except for actor dummy objects, where the object
+   *  must be generated by executing the task locally. */
+  bool request_transfer;
```

Comment:
Ah oops, that was my bad. It's supposed to check the field in `fetch_object_timeout_handler`. I'll fix that and see if I can come up with a test case for it as well.

The actor dummy objects are only supposed to be generated by the actor itself. Basically, the scenario where it would break reconstruction is if a local scheduler dies, but the corresponding plasma manager is still reachable. Then, when reconstructing the dummy object, if we transfer the object from the surviving plasma manager, it will look like we executed that task, even though we didn't. I'll clarify the documentation for that scenario.